Makefile

```diff
@@ -1,4 +1,4 @@
-CC=gcc
+CC?=gcc
```

Comment:
when I specify clang in travis-ci, lua-cjson, for example just picks up $CC and compiles with clang. I do not call "make CC=$CC" in lua-cjson